sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -738,6 +738,31 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     val optimized = Optimize.execute(query)
     comparePlans(optimized, query)
   }
+
+  test("SPARK-35972: When replace ExtractValue we should use semanticEquals") {
```

Comment:
let's make the test name more general: `NestedColumnAliasing should consider semantic equality`

Reply:
+1